From fb1b353d2e1045b3210200b7ab09f1bd17bab795 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 26 Aug 2014 15:30:26 -0400
Subject: [ethernet] igbvf: add missing iounmap() on error in igbvf_probe()

Message-id: <1409067034-17415-6-git-send-email-sassmann@redhat.com>
Patchwork-id: 89217
O-Subject: [RHEL7.1 PATCH 05/13] igbvf: add missing iounmap() on error in igbvf_probe()
Bugzilla: 1091122
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: John Greene <jogreene@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: John Linville <linville@redhat.com>

From: Wei Yongjun <yongjun_wei@trendmicro.com.cn>

Add the missing iounmap() before return from igbvf_probe()
in the error handling case.

Signed-off-by: Wei Yongjun <yongjun_wei@trendmicro.com.cn>
Tested-by: Aaron Brown <aaron.f.brown@intel.com>
Tested-by: Sibai Li <Sibai.li@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit de524681f88ff4ed293aa239f83c8cb04d59b47d)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/igbvf/netdev.c b/drivers/net/ethernet/intel/igbvf/netdev.c
index 0c0b8f6..04bf22e 100644
--- a/drivers/net/ethernet/intel/igbvf/netdev.c
+++ b/drivers/net/ethernet/intel/igbvf/netdev.c
@@ -2692,7 +2692,7 @@ static int igbvf_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
  if (ei->get_variants) {
   err = ei->get_variants(adapter);
   if (err)
-   goto err_ioremap;
+   goto err_get_variants;
  }
 
  /* setup adapter struct */
@@ -2789,6 +2789,7 @@ err_hw_init:
  kfree(adapter->rx_ring);
 err_sw_init:
  igbvf_reset_interrupt_capability(adapter);
+err_get_variants:
  iounmap(adapter->hw.hw_addr);
 err_ioremap:
  free_netdev(netdev);
-- 
1.7.1